test/e2e/basic_test.go

```diff
@@ -365,13 +365,41 @@ func testReconcileGatewayRoutesOnStartup(t *testing.T, data *TestData, isIPv6 bo
 		t.Fatalf("Failed to detect gateway interface name from ConfigMap: %v", err)
 	}
 
-	expectedRtNumMin, expectedRtNumMax := clusterInfo.numNodes-1, clusterInfo.numNodes-1
+	nodeNums := clusterInfo.numNodes
+	var expectedRtNum int
 	switch encapMode {
+	case config.TrafficEncapModeEncap:
+		expectedRtNum = nodeNums - 1
+		t.Logf("Encap mode: expected %d gateway routes (one per peer Node)", expectedRtNum)
+
 	case config.TrafficEncapModeNoEncap:
-		expectedRtNumMin, expectedRtNumMax = 0, 0
+		expectedRtNum = 0
+		t.Log("NoEncap mode: expected 0 gateway route")
 
 	case config.TrafficEncapModeHybrid:
-		expectedRtNumMin = 1
+		currentNodeIPStr := nodeIPv4(0)
+		if isIPv6 {
+			currentNodeIPStr = nodeIPv6(0)
+		}
+
+		podName := getAntreaPodName(t, data, nodeName)
+		cmd := []string{"sh", "-c", fmt.Sprintf("ip -o addr show | grep %s | awk '{print $4}' | tr -d ' \\n'", currentNodeIPStr)}
+		stdout, _, err := data.RunCommandFromPod(antreaNamespace, podName, agentContainerName, cmd)
```

Comment:
can you move this to a separate helper function

additionally, I think that as a rule it is better to:
- avoid wrapping commands in a shell if possible
- limit the dependency on tools (grep, awk, tr) by running the most basic command in the Pod (e.g., `ip addr show`) and then parsing the output programmatically from the test code.

In this case, I think the existing `GetTransportInterface` function can be used as a reference (and maybe there is some possibility of code unification as well).